jerry-ext/debugger/debugger-tcp.c

```diff
@@ -51,6 +52,7 @@ typedef SOCKET jerryx_socket;
 
 /* On *nix the sockets are integer identifiers */
 typedef int jerryx_socket;
+typedef size_t jerryx_socket_size_t;
```

Comment:
On Windows, `send` and `recv` have the following signatures:
`int send(SOCKET s, const char *buf, int len, int flags);`
`int recv(SOCKET s, char *buf, int len, int flags);`

On Linux/POSIX, `send` and `recv` look like:
`ssize_t send(int sockfd, const void *buf, size_t len, int flags);`
`ssize_t recv(int sockfd, void *buf, size_t len, int flags);`

I think it would be useful to have typedefs for the first three arguments, as they are all different on the two platforms. It should look something like the following:

```c
/*
 * On Windows, send has the following signature:
 * int send(SOCKET s, const char *buf, int len, int flags);
 * int recv(SOCKET s, char *buf, int len, int flags);
 */
typedef SOCKET jerryx_socket_t; 
typedef char *jerryx_socket_buffer_t; 
typedef int jerryx_socket_buffer_length_t;
```

```c
/*
 * On *nix, send and recv have the following signatures:
 * ssize_t send(int sockfd, const void *buf, size_t len, int flags);
 * ssize_t recv(int sockfd, void *buf, size_t len, int flags);
 */
typedef int jerryx_socket_t;
typedef void *jerryx_socket_buffer_t;
typedef size_t jerryx_socket_buffer_length_t;
```

Note that the existing `jerryx_socket` type name has also been changed.

Reply:
Do I need also define the jerryx_socket_return_t to int/ssize_t separately?